src/core_plugins/kbn_vislib_vis_types/public/controls/heatmap_options.js

```diff
@@ -13,7 +13,7 @@ module.directive('heatmapOptions', function () {
       $scope.showLabels = false;
       $scope.customColors = false;
       $scope.options = {
-        rotateLabels: false
+        rotateLabels: $scope.vis.params.valueAxes[0].labels.rotate === 270
```

Comment:
you could also store the actual rotation as true/false, iso deriving it from the numerical value.